ra2ce/network/network_simplification/snkit_network_merge_wrapper.py

```diff
@@ -94,7 +95,92 @@ def _get_merged_edges(
         updated_edges_gdf = updated_edges_gdf.drop(columns=["id"])
         return updated_edges_gdf
 
+    def _filter_degree_2(degree_2s: set) -> set:
+        from ra2ce.network.network_simplification.snkit_network_wrapper import SnkitNetworkWrapper
+
+        degree_2_filtered = set()
+        snkit_network_wrapper = SnkitNetworkWrapper(
+            snkit_network=snkit_network,
+        )
+        networkx_graph = snkit_network_wrapper.to_networkx()
+        for degree_2_node_id in degree_2s:
+            # Get the predecessors (antecedents) and successors (precedents)
+            predecessors = list(networkx_graph.predecessors(degree_2_node_id))
+            successors = list(networkx_graph.successors(degree_2_node_id))
+
+            # Check if there is exactly one predecessor and one successor: 1->2, 2->3, 2->1, 3->2 => filters on 2
+            if len(predecessors) == len(successors) == 1:
+                degree_2_filtered.add(degree_2_node_id)
+        return degree_2_filtered
+
+    def _filter_degree_4(degree_4s: set) -> set:
+        from ra2ce.network.network_simplification.snkit_network_wrapper import SnkitNetworkWrapper
+
+        degree_4_filtered = set()
+        snkit_network_wrapper = SnkitNetworkWrapper(
+            snkit_network=snkit_network,
+        )
+        networkx_graph = snkit_network_wrapper.to_networkx()
+        for degree_4_node_id in degree_4s:
+            # Get the predecessors (antecedents) and successors (precedents)
+            predecessors = list(networkx_graph.predecessors(degree_4_node_id))
+            successors = list(networkx_graph.successors(degree_4_node_id))
+
+            # Check if there is exactly one predecessor and one successor: 1->2, 2->3, 2->1, 3->2 => filters on 2
+            if sorted(predecessors) == sorted(successors):
+                degree_4_filtered.add(degree_4_node_id)
+        return degree_4_filtered
+
     def _get_edge_paths(node_set: set, snkit_network: SnkitNetwork) -> list:
+        def _find_and_append_degree_4_paths(
```

Comment:
Add a proper return to the function instead of adjusting the input.

Reply:
Done